ldcomponents/data_system_configuration_builder.go

```diff
@@ -0,0 +1,199 @@
+package ldcomponents
+
+import (
+	"errors"
+	"fmt"
+
+	ss "github.com/launchdarkly/go-server-sdk/v7/subsystems"
+)
+
+// DataSystemConfigurationBuilder is a builder for configuring the SDK's data acquisition strategy.
+type DataSystemConfigurationBuilder struct {
+	storeBuilder         ss.ComponentConfigurer[ss.DataStore]
+	storeMode            ss.DataStoreMode
+	initializerBuilders  []ss.ComponentConfigurer[ss.DataInitializer]
+	primarySyncBuilder   ss.ComponentConfigurer[ss.DataSynchronizer]
+	secondarySyncBuilder ss.ComponentConfigurer[ss.DataSynchronizer]
+	config               ss.DataSystemConfiguration
+}
+
+// Endpoints represents custom endpoints for LaunchDarkly streaming and polling services.
+//
+// You may specify none, one, or both of these endpoints via WithEndpoints. If an endpoint isn't specified,
+// then the default endpoint for that service will be used.
+//
+// This is a convenience that is identical to individually configuring polling or streaming synchronizer
+// BaseURI's using their specific builder functions.
+//
+// To specify Relay Proxy endpoints, use WithRelayProxyEndpoints.
+type Endpoints struct {
+	Streaming string
+	Polling   string
+}
+
+// DataSystemModes provides access to high level strategies for fetching data. The default mode
+// is suitable for most use-cases.
+type DataSystemModes struct {
+	endpoints Endpoints
+}
+
+// Default is LaunchDarkly's recommended flag data acquisition strategy. Currently, it operates a
+// two-phase method for obtaining data: first, it requests data from LaunchDarkly's global CDN. Then, it initiates
+// a streaming connection to LaunchDarkly's Flag Delivery services to receive real-time updates. If
+// the streaming connection is interrupted for an extended period of time, the SDK will automatically fall back
+// to polling the global CDN for updates.
+func (d *DataSystemModes) Default() *DataSystemConfigurationBuilder {
+	streaming := StreamingDataSourceV2()
+	if d.endpoints.Streaming != "" {
+		streaming.BaseURI(d.endpoints.Streaming)
```

Comment:
Maybe we need to set BaseURI internally to the data source, otherwise someone might forget to call BaseURI.